### DISCUSSION TITLE
How to reset other fields except File in react hook form ?

### DISCUSSION THREAD

**Discussion by ShawUz** [react-hook-form/react-hook-form#4887] [Q&A]
I have a requirement that the form contains following field :
   { 
    reasonCode: '',
    subReasonCode: '',
    reasonDetail: '',
    file: null, 
   }

The file field is implemented by 
  <input
        ref={register({ required: true })}
        type="file"
        accept=".csv,.xls,.xlsx"
        style={{ display: 'none' }}
        name="file"
   />

I want to clear subReasonCode and reasonDetail when reasonCode is changed, but when I try to call
     reset({
      ...getValues(),
      subReasonCode: '',
      reasonDetail: '',
    });
the file field is also cleared. 

Anyone has idea how to solve this problem?

Thanks.

**Comment by bluebill1049** [comment]
`File` is a tricky one, as it related to security. I would probably try to use setValue instead to those other fields.

**Reply by ShawUz** [reply]
Thanks for your reply.

A problem is that setValue will not trigger re-render. There is a cascade dropdown in this form, i.e. reasonCode and subReasonCode field. So I need to clear the subReasonCode filed when the option of reasonCode is changed. Is there a way to implement this ?
